backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxSparkPlanExecApi.scala

```diff
@@ -43,4 +44,28 @@ class VeloxSparkPlanExecApi extends GlutenSparkPlanExecApi {
                                          attributeSeq: Seq[Attribute]): ExpressionTransformer = {
     new GlutenNamedStructTransformer(substraitExprName, original, attributeSeq)
   }
+
+  /**
+   * To align with spark in casting string type input to integral type,
+   * add trim node for trimming whitespace. See spark's toInt in UTF8String.java.
+   */
+  override def genCastWithNewChild(c: Cast): Cast = {
+    // Whitespace to be trimmed, including: ' ', '\n', '\r', '\f', etc.
+    // The space separator (except ' ', i.e., '\u0020'), line separator &
+    // paragraph separator are not supported to trim, assuming they are rarely used.
+    val trimStr = " \t\n\u000B\u000C\r\u001C\u001D\u001E\u001F"
+    c.dataType match {
+      case ByteType | ShortType | IntegerType | LongType =>
```

Comment:
After check the `Cast` in Spark, I found the `Character.isWritespace` is also used while casting `String` to `TimeStamp, Date, YearMonthInterval, DayTimeInterval, CalendarInterval, Boolean, TimestampNTZ`. As for casting from `String` to `Double, Float, Decimal`, Spark use `String.trim`.
May be this code is better?
```
c.dataType match {
  case _: ArrayType | _: MapType | _: StructType | _: UserDefinedType[_] =>
     c
  case FloatType | DoubleType | DecimalType =>
    c.child.dataType match {
      case StringType =>
        val trim = StringTrim(c.child, Some(Literal(trimSpaceStr)))
          c.withNewChildren(Seq(trim)).asInstanceOf[Cast]
       case _ =>
          c
      }
    }
  case _: =>
    c.child.dataType match {
      case StringType =>
        val trim = StringTrim(c.child, Some(Literal(trimStr)))
          c.withNewChildren(Seq(trim)).asInstanceOf[Cast]
        case _ =>
          c
      }
    }
```

Reply:
Thanks so much for your check! Yes, we should also consider all other types even though the potential cast issues have not been exposed yet.

Reply:
For BinaryType, we should keep it not trimmed.

Reply:
Ah, yes